Index: libdicl-0.1.39/gl/getprogname.c
===================================================================
--- libdicl-0.1.39.orig/gl/getprogname.c
+++ libdicl-0.1.39/gl/getprogname.c
@@ -49,6 +49,9 @@
 # include <stdio.h>
 # include <fcntl.h>
 # include <sys/procfs.h>
+
+// magic from rld
+extern const char** __Argv;
 #endif
 
 #include "dirname.h"
@@ -219,9 +222,13 @@ getprogname (void)
     }
   return p;
 # elif defined __sgi                                        /* IRIX */
+  static char *progname = 0;
   char filename[50];
   int fd;
 
+  if (progname != 0)
+    return progname;
+
   sprintf (filename, "/proc/pinfo/%d", (int) getpid ());
   fd = open (filename, O_RDONLY);
   if (0 <= fd)
@@ -240,11 +247,22 @@ getprogname (void)
           if (namecopy)
             {
               namecopy[namelen] = '\0';
-              return memcpy (namecopy, name, namelen);
+              progname = (char*) memcpy (namecopy, name, namelen);
             }
         }
     }
-  return NULL;
+  else if (__Argv) {
+    const char* argv0 = __Argv[0];
+    const char* slash = strrchr(argv0, '\\');
+    if (slash) {
+      progname = (char*) malloc (strlen(slash));
+      strcpy(progname, slash + 1);
+    } else {
+      progname = strdup(argv0);
+    }
+  }
+
+  return progname;
 # else
 #  error "getprogname module not ported to this OS"
 # endif
